### PR TITLE
add parameter to set cadl emitter options

### PR DESCRIPTION
**Pull request by chunyu3**
Fix https://github.com/Azure/azure-sdk-for-net/issues/34357

- Add parameter to set cadl emitter options

### CODE REVIEW

File: eng/common/scripts/Cadl-Project-Generate.ps1 [hunk 1]
```diff
@@ -4,7 +4,9 @@
 param (
     [Parameter(Position=0)]
     [ValidateNotNullOrEmpty()]
-    [string] $ProjectDirectory
+    [string] $ProjectDirectory,
+    [Parameter(Position=1)]
+    [string] $CadlAdditionalOptions ## addtional cadl emitter options, separated by semicolon if more than one
```
Comment (m-nash) [inline eng/common/scripts/Cadl-Project-Generate.ps1:9]: Can we modify this to accept just `key=value;key=value` and not require the user to pass in the emitter name?  We know this from `$emitterName` and can construct this part for the customer to simplify their command line call.

Comment (chunyu3) [reply eng/common/scripts/Cadl-Project-Generate.ps1]: updated

**Comment by weshaggard** [comment]
@chunyu3 See [eng/common workflow](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/README.md#workflow) for the steps to merged eng/common changes. If you still need help let me know.

**Comment by chunyu3** [comment]
> the steps to merged eng/common changes. If you still need help let me know.

**Comment by chunyu3** [comment]
@weshaggard The cpp CI pipeline failed, https://github.com/Azure/azure-sdk-for-cpp/pull/4396/checks?check_run_id=11673492430, who can I ask to help on this? Thanks

**Review by weshaggard — APPROVED** [review]

**Pull request merged**